vm/src/stdlib/os.rs

```diff
@@ -1697,6 +1697,42 @@ pub(super) mod _os {
         }
     }
 
+    // TODO: libc currently doesn't support nl_langinfo in wasi
+    #[cfg(not(target_os = "wasi"))]
+    #[pyfunction]
+    fn device_encoding(fd: i32, _vm: &VirtualMachine) -> PyResult<Option<String>> {
+        if !isatty(fd) {
+            return Ok(None);
+        }
+
+        cfg_if::cfg_if! {
+            if #[cfg(target_os = "android")] {
+                Ok(Some("UTF-8".to_owned()))
+            } else if #[cfg(windows)] {
+                let cp = if fd == 0 {
```

Comment:
Here is warnings. I think `match` fits here better becaues all of the condition expression is about `fd`

Reply:
Changed as requested in new commit.